lib/logstash/outputs/redis.rb

```diff
@@ -185,7 +188,13 @@ def congestion_check(key)
     return if @congestion_threshold == 0
     if (Time.now.to_i - @congestion_check_times[key]) >= @congestion_interval # Check congestion only if enough time has passed since last check.
       while @redis.llen(key) > @congestion_threshold # Don't push event to Redis key which has reached @congestion_threshold.
-        @logger.warn? and @logger.warn("Redis key size has hit a congestion threshold #{@congestion_threshold} suspending output for #{@congestion_interval} seconds")
+        if @shuffle_hosts
+          # Switch to different server, and try again.
+          @logger.warn? and @logger.warn("Redis key size has hit a congrestion threshold #{@congestion_threshold} switching to next server.")
+          @redis = connect
+        else
+          @logger.warn? and @logger.warn("Redis key size has hit a congestion threshold #{@congestion_threshold} suspending output for #{@congestion_interval} seconds")
```

Comment:
Contextual logging (A message and the related data) should have the message and data be totally separate. For example:

```ruby
@logger.warn? && @logger.warn("Redis key size has hit congestion threshold. Suspending output temporarily", :threshold => @congestion_threshold, :interval => @congestion_interval)
```